cmake/OmimHelpers.cmake

```diff
@@ -137,3 +147,93 @@ function(add_clang_compile_options)
     add_compile_options(${ARGV})
   endif()
 endfunction()
+
+function(export_directory_flags _filename)
+  get_directory_property(_include_directories INCLUDE_DIRECTORIES)
+  get_directory_property(_cmp_definitions COMPILE_DEFINITIONS)
+  get_directory_property(_cmp_flags COMPILE_FLAGS)
+  get_directory_property(_cmp_options COMPILE_OPTIONS)
+
+  # Append Release/Debug flags
+  string(TOUPPER "${CMAKE_BUILD_TYPE}" upper_build_type)
+  set(_cmp_flags "${CMAKE_CXX_FLAGS_${upper_build_type}}")
+
+  set(_include_directories "$<$<BOOL:${_include_directories}>:-I$<JOIN:${_include_directories},\n-I>\n>")
+  set(_cmp_definitions "$<$<BOOL:${_cmp_definitions}>:-D$<JOIN:${_cmp_definitions},\n-D>\n>")
+  set(_cmp_flags "$<$<BOOL:${_cmp_flags}>:$<JOIN:${_cmp_flags},\n>\n>")
+  set(_cmp_options "$<$<BOOL:${_cmp_options}>:$<JOIN:${_cmp_options},\n>\n>")
+  file(GENERATE OUTPUT ${_filename} CONTENT "${_cmp_definitions}${_include_directories}${_cmp_flags}${_cmp_options}\n")
+endfunction()
+
+function(add_precompiled_header header pch_target_name)
+  set(pch_flags_file "${CMAKE_BINARY_DIR}/${pch_target_name}_flags_file")
+  export_directory_flags("${pch_flags_file}")
+  set(compiler_flags "@${pch_flags_file}")
+  # CMAKE_CXX_STANDARD 14 flags
+  set(c_standard_flags "-std=c++14" "-std=gnu++14")
+  file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/pch_lib")
+  file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/pch_exe")
+  file(COPY "${header}" DESTINATION "${CMAKE_BINARY_DIR}/pch_lib")
+  file(COPY "${header}" DESTINATION "${CMAKE_BINARY_DIR}/pch_exe")
+
+  get_filename_component(pch_file_name ${header} NAME)
+  set(lib_header "${CMAKE_BINARY_DIR}/pch_lib/${pch_file_name}")
+  set(exe_header "${CMAKE_BINARY_DIR}/pch_exe/${pch_file_name}")
+
+  set(compiled_lib_header "${CMAKE_BINARY_DIR}/pch_lib/${pch_file_name}.${PCH_EXTENSION}")
+  set(compiled_exe_header "${CMAKE_BINARY_DIR}/pch_exe/${pch_file_name}.${PCH_EXTENSION}")
+
+  add_custom_target(
+    "${pch_target_name}_lib"
+    COMMAND "${CMAKE_CXX_COMPILER}" ${compiler_flags} ${c_standard_flags} "-fPIC" -x c++-header -c "${lib_header}" -o "${compiled_lib_header}"
+    COMMENT "Building precompiled omim CXX header for libs"
+  )
+  add_custom_target(
+    "${pch_target_name}_exe"
+    COMMAND "${CMAKE_CXX_COMPILER}" ${compiler_flags} ${c_standard_flags} "-fPIE" -x c++-header -c "${exe_header}" -o "${compiled_exe_header}"
+    COMMENT "Building precompiled omim CXX header for executables"
+  )
+  add_custom_target(
+    "${pch_target_name}"
+    COMMENT "Wait both lib/exe precompiled header built"
```

Comment:
Waiting for both lib and exe precompiled headers
Waiting for both lib and exe precompiled headers to build

Reply:
Fixed.